x.py

```diff
@@ -4,6 +4,21 @@
 
 import os
 import sys
+
+# If this is python2, check if python3 is available and re-execute with that
+# interpreter.
+if sys.version_info.major < 3:
+    try:
+        # On windows, `py -3` sometimes works
+        # Try this first, because 'python3' can act weird on windows
```

Comment:
Can we be more specific about the behavior here? Something like "python3 on windows appears to implicitly launch the app store in some cases, which is not the desired behavior for us"

Reply:
Done.